crates/node-db/tests/insert.rs

```diff
@@ -189,9 +194,12 @@ fn test_finalize_block() {
         NUM_FINALIZED_BLOCKS - 1,
         Duration::from_secs((NUM_FINALIZED_BLOCKS + 1000) as u64),
     );
-    let tx = conn.transaction().unwrap();
-    node_db::insert_block(&tx, &fork).unwrap();
-    tx.commit().unwrap();
+
+    node_db::with_tx::<_, QueryError>(&mut conn, |tx| {
+        node_db::insert_block(tx, &fork).unwrap();
+        Ok(())
+    })
+    .unwrap();
```

Comment:
Rust can infer the error type if we return a result from the body of the function, e.g.
```suggestion
    node_db::with_tx(&mut conn, |tx| node_db::insert_block(tx, &fork)).unwrap();
```
It might be worth going over the other changes to see if any others can be simplified by using inference like this